cmd/panfs-multipart.go

```diff
@@ -154,25 +179,41 @@ func (fs *PANFSObjects) backgroundAppend(ctx context.Context, bucket, object, up
 			break
 		}
 		sort.Strings(entries)
-
 		for _, entry := range entries {
-			if entry == fs.metaJSONFile {
-				continue
-			}
 			partNumber, etag, actualSize, err := fs.decodePartFile(entry)
 			if err != nil {
 				// Skip part files whose name don't match expected format. These could be backend filesystem specific files.
 				continue
 			}
 			if partNumber < nextPartNumber {
 				// Part already appended.
+				requiredAppendFileSize += actualSize
 				continue
 			}
 			if partNumber > nextPartNumber {
 				// Required part number is not yet uploaded.
 				break
 			}
 
+			if requiredAppendFileSize != actualAppendFileSize {
+				// actual append file size is not match to parts sizes from mparts.json -> reset all
+				if err = fsRemoveFile(ctx, file.filePath); err != nil {
+					return
+				}
+				if err = fsRemoveFile(ctx, file.partsInfoFile); err != nil {
+					return
+				}
+				fsParts.Parts = nil
+				initBGAppend()
+				if resetCount >= resetLimit {
```

Comment:
What purpose do `resetCount` and `resetLimit` serve here?

Reply:
If for some reason encoded file size (that is part of file part name) is not match to the real file size then the bgAppend loop will never ends (because the sum of decoded parts sizes will not match to append file size) so the reset count is a just a way how to break the loop (however it does not seems tb be perfect solution for me..)

In any way this is almost impossible case - I cannot imagine how we can reach such situation when some part will be written with wrong decoded size in file name

Reply:
The code
```go
				if resetCount >= resetLimit {
					// If for some reason decoded part size is not match to real part size then we will never reach
					// the requiredAppendFileSize == actualAppendFileSize condition. So set the limit of attempts for reset
					break
				}
```
breaks out of the inner loop which is just iterating over a list of entries read on line 175.
We're going to get out of the outer loop when we discover the upload directory has not been modified since the last iteration:
```go
		if uploadIDDirStat.ModTime() == prevUploadIDDirStat.ModTime() {
			break
		}
```

I think we're safe here even without `resetCount` and `resetLimit`. Please, correct me if I missed something.

Reply:
Yes, you are right. I think even that calculating of "requiredAppendFileSize"is redundant in `bgAppend` as such validation exists in complete multipart method